lib/modules/manager/flux/extract.ts

```diff
@@ -102,6 +106,39 @@ function resolveGitRepositoryPerSourceTag(
   }
 }
 
+function resolveHelmRepository(
```

Comment:
same, extract to refactor pr. can combined with comment above

Reply:
sorry, but I cannot really follow this. the extracted code in this function is _literally_ identical to what was previously within the switch statement, and I need the _exact_ same logic in the new case (as it's just about resolving the details of the referenced repository, properly supporting OCI repos etc). If I don't extract this into a function, I'd have to duplicate the whole block, copy+paste it down into the new case. which causes test coverage to decrease, and forcing the addition (and duplication) of test cases just to test the same code again in the other context - and to remove all of these additional tests again right afterwards in yet another PR too? that doesn't make sense.

I've removed the other refactoring as requested though, as that was in a part that is now mostly unchanged again after working on the various topics. but I really don't see how the second one here could be avoided without creating more questionable quality issues.